frontend/src/components/Module/ModuleDetailsContractLinkEvents.vue

```diff
@@ -4,7 +4,9 @@
 			<TableTh>Transaction</TableTh>
 			<TableTh>Contract Address</TableTh>
 			<TableTh>Age</TableTh>
-			<TableTh>Action</TableTh>
+			<TableTh>Action
+				<InfoTooltip text="Action which defines if a contract was linked (added) or unlinked (removed) to the module execution code."/>							
```

Comment:
I don't understand this concept. An instance has a code pointer (to the module) at all times. That can be changed by the upgrade call. I don't understand which events this is describing.

Reply:
When a contracts upgrades by pointing to module A instead of B two events will be created. One with action removed on module B and one with action added on module A.

Reply:
But why? This is an atomic action and only one event is generated on the chain. Why do we need to display two and introduce new concepts?

Reply:
Atomic operations can generate multiple events and the data model of the CCD scan shouldn't be a 1-1 representation of the chains data model. The CCD scan’s purpose is to make a projection of the chains state such that the state can be interpreted easily by the users.

It makes sense to show to the users when a contract pointer was removed from one module. If the current table and events wasn’t present the user would need to click a lot around to get the same information.

If you have more questions to the UI and its components you are welcome to come over :-) I don’t think continuing the discussion here will be efficient.